File: 백준/Bronze/4084. Viva la Diferencia/Viva la Diferencia.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        while (true) {
            int[] numbers = Arrays.stream(br.readLine().split(" "))
                    .mapToInt(Integer::parseInt)
                    .toArray();

            if (numbers[0] == 0 && numbers[1] == 0 && numbers[2] == 0 && numbers[3] == 0)
                break;

            int count = 0;

            while (true) {
                int a = numbers[0];
                int b = numbers[1];
                int c = numbers[2];
                int d = numbers[3];

                if (a == b && b == c && c == d)
                    break;

                numbers[0] = Math.abs(a - b);
                numbers[1] = Math.abs(b - c);
                numbers[2] = Math.abs(c - d);
                numbers[3] = Math.abs(d - a);

                count++;
            }

            System.out.println(count);
        }
    }
}

File: 백준/Bronze/4084. Viva la Diferencia/Viva la Diferencia.py
while True:
    a, b, c, d = map(int, input().split())
    if not a and not b and not c and not d:
        break

    numbers = [a, b, c, d]
    count = 0

    while any(numbers[0] != x for x in numbers):
        a, b, c, d = numbers
        numbers = [abs(a - b), abs(b - c), abs(c - d), abs(d - a)]
        count += 1

    print(count)